server/handlers/level_handler.coffee

```diff
@@ -80,6 +81,7 @@ LevelHandler = class LevelHandler extends Handler
         valueArray = _.pluck data, (session) -> _.find(session.leagues, leagueID: league)?.stats?.totalScore or 10
       else
         valueArray = _.pluck data, 'totalScore'
+        simpleCache.setLadderScores req.query.levelSlug, match.team, valueArray
```

Comment:
One idea for improving this: round scores to the nearest 100 or some other interval of points, and return sums. Can definitely still render the histogram with this, and get a rough rank. Might also improve performance by decreasing the amount of data aggregated and returned.

Reply:
Good idea–that would work for the original use of the histograms, but then I wouldn't be able to use it for the other two functions I'm now using it for. Ladder histograms have seemed to be okay with the aggressive caching we use, even though those request are pretty long.